src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,370 @@
+      logger.info("Error reading envelope");
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+    }
+
+    callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
+  }
+
+  private void setUser(final JSONObject jsonUser, final JSONObject otherUserKeys, final CallbackContext callbackContext) {
+    Sentry.configureScope(scope -> {
+      try {
+        if (jsonUser == null && otherUserKeys == null) {
+          scope.setUser(null);
+        } else {
+          User userInstance = new User();
+
+          if (jsonUser != null) {
+            if (jsonUser.has("email")) {
+              userInstance.setEmail(jsonUser.getString("email"));
+            }
+
+            if (jsonUser.has("id")) {
+              userInstance.setId(jsonUser.getString("id"));
+            }
+
+            if (jsonUser.has("username")) {
+              userInstance.setUsername(jsonUser.getString("username"));
+            }
+
+            if (jsonUser.has("ip_address")) {
+              userInstance.setIpAddress(jsonUser.getString("ip_address"));
+            }
+          }
+
+          if (otherUserKeys != null) {
+            HashMap<String, String> otherUserKeysMap = new HashMap<String, String>();
+            Iterator<String> it = otherUserKeys.keys();
+            while (it.hasNext()) {
+              String key = it.next();
+              String value = otherUserKeys.getString(key);
+
+              otherUserKeysMap.put(key, value);
+            }
+
+            userInstance.setOthers(otherUserKeysMap);
+          }
+
+          scope.setUser(userInstance);
+        }
+      } catch (JSONException e) {
+        logger.info("Error deserializing user");
+      }
+    });
+
+    callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
+  }
+
+  private void addBreadcrumb(final JSONObject jsonBreadcrumb, final CallbackContext callbackContext) {
+    Sentry.configureScope(scope -> {
+      try {
+        Breadcrumb breadcrumb = new Breadcrumb();
+
+        if (jsonBreadcrumb.has("message")) {
+          breadcrumb.setMessage(jsonBreadcrumb.getString("message"));
+        }
+
+        if (jsonBreadcrumb.has("type")) {
+          breadcrumb.setType(jsonBreadcrumb.getString("type"));
+        }
+
+        if (jsonBreadcrumb.has("category")) {
+          breadcrumb.setCategory(jsonBreadcrumb.getString("category"));
+        }
+
+        if (jsonBreadcrumb.has("level")) {
+          breadcrumb.setLevel(getSentryLevelFromString(jsonBreadcrumb.getString("level")));
+        }
+
+        if (jsonBreadcrumb.has("data")) {
+          JSONObject data = jsonBreadcrumb.getJSONObject("data");
+          Iterator<String> it = data.keys();
+          while (it.hasNext()) {
+            String key = it.next();
+            String value = data.getString(key);
+
+            breadcrumb.setData(key, value);
+          }
+        }
+
+        scope.addBreadcrumb(breadcrumb);
+        logger.info("Send breadcrumb successful");
```

Comment:
```suggestion
        logger.debug("add breadcrumb successful");
```

Reply:
there's no `.debug` [on logger](https://developer.android.com/reference/java/util/logging/Logger) (to my surprise as well)